src/scripts/main.js

```diff
@@ -1,3 +1,72 @@
 'use strict';
 
-// write code here
+const table = document.querySelector('table');
+const rows = table.rows;
+
+const appendRow = document.querySelector('.append-row');
+const removeRow = document.querySelector('.remove-row');
+const appendColumn = document.querySelector('.append-column');
+const removeColumn = document.querySelector('.remove-column');
+
+const minCells = 2;
+const maxCells = 10;
+
+appendRow.onclick = function () {
+  if (rows.length < maxCells) {
+    const newRow = table.insertRow();
+
+    const columnCount = rows[0].cells.length;
+
+    for (let i = 0; i < columnCount; i++) {
+      newRow.insertCell(-1);
+    }
+
+    removeRow.disabled = false;
+
+    if (rows.length === maxCells) {
+      appendRow.disabled = true;
+    }
+  }
+};
```

Comment:
Code Duplication Issue: The logic for appending and removing rows is very similar to the logic for appending and removing columns. Consider creating a separate function for these similar operations to adhere to the DRY (Don't Repeat Yourself) principle. This will make your code cleaner and easier to maintain.